sdk/test-utils/test-utils-vitest/src/tracing/testSpan.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  Span,
+  AttributeValue,
+  Attributes,
+  SpanContext,
+  SpanKind,
+  Tracer,
+  TimeInput,
+  SpanStatus,
+  SpanStatusCode,
+  Link,
+} from "@opentelemetry/api";
+/**
+ * A mock span useful for testing.
+ */
+export class TestSpan implements Span {
+  /**
+   * The Span's current name
+   */
+  name: string;
+
+  /**
+   * The Span's current status
+   */
+  status: SpanStatus;
+
+  /**
+   * The Span's kind
+   */
+  kind: SpanKind;
+
+  /**
+   * True if end() has been called on the Span
+   */
+  endCalled: boolean;
+
+  /**
+   * The start time of the Span
+   */
+  readonly startTime: TimeInput;
+
+  /**
+   * The id of the parent Span, if any.
+   */
+  readonly parentSpanId?: string;
+
+  /**
+   * The Span's links.
+   */
+  readonly links: Link[];
+
+  /**
+   * Known attributes, if any.
+   */
+  readonly attributes: Attributes;
+
+  private _context: SpanContext;
+  private readonly _tracer: Tracer;
+
+  /**
+   * Starts a new Span.
+   * @param parentTracer-  The tracer that created this Span
```

Comment:
interesting that linter is happy without proper spacing and prettier doesn't help?
```suggestion
   * @param parentTracer - The tracer that created this Span
```